### ISSUE TITLE
Feature request: Clean logging

### ISSUE THREAD

**Issue by dr460neye** [fjogeleit/policy-reporter#273]
At the moment, there are barely logs available. This is one of the reasons which makes it hard for me to setup and configure the policy reporter and additional components.

For this i would like to request:

- Adding improved logs, by using a logging library
- In best case this supports json as an alternative logging format
- Adding log levels
- Support HTTP access logs e.g. to identify your upstream problem
- Add logs for "finished bootup"

Any plans for such a feature already?

**Comment by fjogeleit** [comment]
Hey, thanks for your request. Currently there where no plans for it but let me see what I can do for a first improvement.

**Comment by dr460neye** [comment]
Would really help me a lot :) In specific the upstream info to identify where the firewall or routing raises the problems in accessing the ui.

**Comment by fjogeleit** [comment]
so, the problem you facing is accessing the UI? Over an ingress? Or you see the UI but the API returning errors?

**Comment by dr460neye** [comment]
As there are no logs its quite hard to figure out the source.

When i use the ui via a port-forward, the api calls are passing a 301 move, then get forwarded to url//path (there is somehow a second / appearing.

As the policy reporter does not log at all, i am not sure if it runs well, as the health checks are not passing.
So the problems could be between:
- misconfigured helm chart / application
- over firewall rules
- over ingress configs

Currently for me it does not make sense to investigate further until there are logs :) But happy to share the outcome with the next version which is providing some more logs in the tool stack.

**Comment by dr460neye** [comment]
As i cannot develop something in this tool at the moment, where i can help is to point out where to make some quickwins in the improvement, if this helps:


https://github.com/kyverno/policy-reporter/blob/175770440273ccc753acdd8bdd7da382d75fc779/pkg/api/server.go#L84
-> Add a LoggingHandler for stdout

Happy to look for more if you believe this helps

**Comment by fjogeleit** [comment]
Okay, thanks. Then I have a hint what kind of logs could help.

I will try to add some logging in the UI API Proxy calls and core API calls

**Comment by fjogeleit** [comment]
A first draft / example for API Request Logging:

It will be optional for the API and can be displayed as json or console:

```yaml
logging:
  encoding: json
#  logLevel: 0
#  development: true

api:
  logging: true
```

JSON:

```bash
{"level":"info","ts":1678897266.231006,"caller":"config/target_factory.go:499","msg":"UI Hook configured"}
{"level":"info","ts":1678897266.2408352,"caller":"cmd/run.go:67","msg":"REST api enabled"}
{"level":"info","ts":1678897266.240923,"caller":"cmd/run.go:73","msg":"metrics enabled"}
{"level":"info","ts":1678897266.241184,"caller":"cmd/run.go:115","msg":"start client","worker":3}
{"level":"info","ts":1678897270.873065,"caller":"api/server.go:134","msg":"Serve","proto":"HTTP/1.1","user-agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36","path":"/metrics"}
{"level":"info","ts":1678897273.808793,"caller":"api/server.go:134","msg":"Serve","proto":"HTTP/1.1","user-agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36","path":"/v1/namespaced-resources/report-labels","query":"sources=Trivy%20Vulnerability"}
{"level":"info","ts":1678897281.180486,"caller":"api/server.go:134","msg":"Serve","proto":"HTTP/1.1","user-agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36","path":"/v1/targets"}
```

Console:

```bash
1.678897657387337e+09   info    Serve   {"proto": "HTTP/1.1", "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36", "path": "/v1/targets"}
```

**Comment by fjogeleit** [comment]
the implementation is merged, in the core as well as UI component, I will let you know when it's released

**Comment by fjogeleit** [comment]
I just released version `2.18.0` of the helm chart with the new logging structure and configuration. I will update the documentation soon. You can have a look at:

https://github.com/kyverno/policy-reporter/blob/main/charts/policy-reporter/values.yaml#L124

to enable access logs and set the logging level to debug and 

https://github.com/kyverno/policy-reporter/blob/main/charts/policy-reporter/values.yaml#L119

to configure the output as JSON

Same goes for the UI Proxy Debug Logging:

https://github.com/kyverno/policy-reporter/blob/main/charts/policy-reporter/charts/ui/values.yaml#L75

Let me know if I. can help you with any issue.
